supla-common/proto.h

```diff
@@ -2064,24 +2246,128 @@ typedef struct {
+ * DEVICE CONFIG STRUCTURES
+ ********************************************/
+
+// SUPLA_DS_CALL_SET_DEVICE_CONFIG
+// SUPLA_SD_CALL_SET_DEVICE_CONFIG
+// Config field should contain fields stored in order as they appear in
+// Fields parameter. Size of parameter depends on Field type.
+typedef struct {
+  unsigned char EndOfDataFlag;  // 1 - last message; 0 - more messages will come
+  unsigned char zero[8];  // for future use
+  unsigned _supla_int64_t Fields;   // bit map of SUPLA_DEVICE_CONFIG_FIELD_
+  unsigned _supla_int16_t ConfigSize;
+  char Config[SUPLA_DEVICE_CONFIG_MAXSIZE];  // Last variable in struct!
+} TSD_SetDeviceConfig;  // v. >= 20
 
+// SUPLA_SD_CALL_SET_DEVICE_CONFIG_RESULT
+// SUPLA_DS_CALL_SET_DEVICE_CONFIG_RESULT
+typedef struct {
+  unsigned char Result;  // SUPLA_CONFIG_RESULT_*
+  unsigned char ConfigType;  // SUPLA_DEVICE_CONFIG_TYPE_*
+  unsigned char zero[8];  // for future use
+} TSD_SetDeviceConfigResult;
+
+
+#define SUPLA_DEVCFG_STATUS_LED_ON_WHEN_CONNECTED  0
+#define SUPLA_DEVCFG_STATUS_LED_OFF_WHEN_CONNECTED 1
+#define SUPLA_DEVCFG_STATUS_LED_ALWAYS_OFF         2
+
+typedef struct {
+  unsigned char StatusLedType;  // SUPLA_DEVCFG_STATUS_LED_
+} TDeviceConfig_StatusLed;  // v. >= 20
+
+typedef struct {
+  unsigned char ScreenBrightness;  // 0-100%
+  unsigned char Automatic;         // 0 - false; 1 - true
+} TDeviceConfig_ScreenBrightness;  // v. >= 20
+
+typedef struct {
+  unsigned char Volume;  // 0-100%
+} TDeviceConfig_ButtonVolume;  // v. >= 20
+
+typedef struct {
+  unsigned char DisableLocalConfig;  // 0 - false (local config enabled)
+                                     // 1 - true (local config disabled)
+} TDeviceConfig_DisableLocalConfig;  // v. >= 20
+
+typedef struct {
+  _supla_int16_t TimezoneOffsetMinutes;  // -1560 .. +1560 minutes
+                                         // biggest timezone diff is 26h
+} TDeviceConfig_TimezoneOffset;  // v. >= 20
+
+typedef struct {
+  unsigned char AutomaticTimeSync;  // 0 - disabled
+                                    // 1 - enabled
+} TDeviceConfig_AutomaticTimeSync;  // v. >= 20
+
+typedef struct {
+  unsigned _supla_int16_t ScreensaverDelayMs;  // delay in ms units
+                                               // 0 - disabled
+} TDeviceConfig_ScreensaverDelay;  // v. >= 20
+
+#define SUPLA_DEVCFG_SCREENSAVER_MODE_OFF          0
+#define SUPLA_DEVCFG_SCREENSAVER_MODE_ALL          1
+#define SUPLA_DEVCFG_SCREENSAVER_MODE_TIME         2
+#define SUPLA_DEVCFG_SCREENSAVER_MODE_MEASUREMENT  3
+
+typedef struct {
+  unsigned char ScreensaverMode;  // SUPLA_DEVCFG_SCREENSAVER_MODE_
+}  TDeviceConfig_ScreensaverMode;  // v. >= 20
+
+
+/********************************************
+ * CHANNEL CONFIG STRUCTURES
+ ********************************************/
+
+#define SUPLA_CONFIG_RESULT_FALSE 0
+#define SUPLA_CONFIG_RESULT_TRUE  1
+#define SUPLA_CONFIG_RESULT_DATA_ERROR 2
+#define SUPLA_CONFIG_RESULT_TYPE_NOT_SUPPORTED 3
```

Comment:
Does this relate to ConfigType?

Reply:
No, those are config Fields definitions. It is draft, since device config is not implemented yet.

Reply:
No, those are generic response codes for "set device config" messages.